test/metabase/query_processor_test/explicit_joins_test.clj

```diff
@@ -726,3 +727,74 @@
                         ["Gizmo"     51 "Gizmo"     2834.88 "Gizmo"     3.64]
                         ["Widget"    54 "Widget"    3109.31 "Widget"    3.15]]
                        (mt/formatted-rows [str int str 2.0 str 2.0] results)))))))))))
+
+(deftest double-quotes-in-join-alias-test
+  (mt/test-drivers (mt/normal-drivers-with-feature :left-join)
+    (testing "Make sure our we handle (escape) double quotes in join aliases. Make sure we prevent SQL injection (#20307)"
+      (let [expected-rows (mt/rows
+                           (mt/run-mbql-query venues
+                             {:joins [{:source-table $$categories
+                                       :alias        "Cat"
+                                       :condition    [:= $id $id]
+                                       :fields       [&Cat.categories.id]}]
+                              :limit 1}))]
+        (is (= 1
+               (count expected-rows)))
+        ;; these normally get ESCAPED by [[metabase.util.honeysql-extensions/identifier]] when they're compiled to SQL,
+        ;; but some fussy databases such as Oracle don't even allow escaped double quotes in identifiers. So make sure
+        ;; that we don't allow SQL injection AND things still work
+        (doseq [evil-join-alias ["users.id\" AS user_id, u.* FROM categories LEFT JOIN users u ON 1 = 1; --"
+                                 "users.id\\\" AS user_id, u.* FROM categories LEFT JOIN users u ON 1 = 1; --"
+                                 "users.id\\u0022 AS user_id, u.* FROM categories LEFT JOIN users u ON 1 = 1; --"
+                                 "users.id` AS user_id, u.* FROM categories LEFT JOIN users u ON 1 = 1; --"
+                                 "users.id\\` AS user_id, u.* FROM categories LEFT JOIN users u ON 1 = 1; --"]]
+          (let [evil-query (mt/mbql-query venues
+                             {:joins [{:source-table $$categories
+                                       :alias        evil-join-alias
+                                       :condition    [:= $id $id]
+                                       :fields       [[:field %categories.id {:join-alias evil-join-alias}]]}]
+                              :limit 1})]
+            (mt/with-native-query-testing-context evil-query
+              (is (= expected-rows
+                     (mt/rows (qp/process-query evil-query)))))))))))
```

Comment:
Just to be clear none of these SQL injections worked before this PR either -- they were always escaped -- but the thought occurred to me that if a driver implementation didn't use our built-in tooling they might accidentally opt themselves out of our escaping. Since @jeff303 is working on tooling to let 3rd party drivers use our test suite I thought adding this extra test made sense. While adding it I discovered #20307 which I was then able to fix as part of this PR